main

```diff
@@ -0,0 +1,120 @@
+#!/usr/bin/env ruby
+require './app'
+
+class Main
+  def menu_options
+    puts ' '
+    puts 'Please choose an option by entering a number:'
+    puts '1 - List all books'
+    puts '2 - List all people'
+    puts '3 - Create a person'
+    puts '4 - Create a book'
+    puts '5 - Create a rental'
+    puts '6 - List all rentals for a given person id'
+    puts '7 - Exit'
+  end
+
+  def add_new_person(app)
+    puts 'Do you want to create a student (1) or a teacher (2)? [Input the number]: '
+    type = gets.chomp.to_i
+    puts 'Name: '
+    name = gets.chomp
+    puts 'Age: '
+    age = gets.chomp.to_i
+    if type == 1
+      puts 'Has parent permission? [Y/N]: '
+      parent_permission = gets.chomp
+      app.create_new_student(age, name, parent_permission: parent_permission.downcase == 'y')
+    else
+      puts 'Specialization: '
+      specialization = gets.chomp
+      app.create_new_teacher(specialization, age, name)
+    end
+
+    puts 'You choose invalid menu, Return to menu'
+  end
+
+  def add_book(app)
+    puts 'Title: '
+    title = gets.chomp
+    puts 'Author: '
+    author = gets.chomp
+    app.create_new_book(title, author)
+  end
+
+  def list_all_books(app)
+    app.list_books
+  end
+
+  def list_all_person(app)
+    app.list_persons
+  end
+
+  def add_new_rental(app)
+    puts 'Books is empity ' if app.books.length.zero?
+    puts 'Person is empity' if app.list_persons.length.zero?
+    puts 'Select a book from the following list by number: '
+
+    app.list_books.each_with_index do |book, index|
+      puts "#{index}) Title: \"#{book.title}\", Author: #{book.author}"
+    end
+
+    book_index = gets.chomp.to_i
+    book = app.books[book_index]
+
+    puts 'Select person from the following list by number:'
+    app.list_persons.each_with_index do |person, index|
+      puts "#{index}) [#{person.class}] Name: #{person.name}, ID: #{person.id}, Age: #{person.age}"
+    end
+    person_index = gets.chomp.to_i
+    person = app.persons[person_index]
+
+    puts 'Date:'
+    date = gets.chomp
+
+    app.create_new_rental(date, person, book)
+  end
+
+  def list_rentals_by_id(app)
+    puts 'Enter person id:'
+    id = gets.chomp.to_i
+    app.list_rentals_by_person_id(id)
+  end
```

Comment:
- Please note that when I tried listing rentals by a `person's id` I get the below error. Kindly fix it 😉
![Screenshot from 2022-08-27 00-35-09](https://user-images.githubusercontent.com/44624138/187003236-0eaca2f4-8726-4335-aec6-4adbf9256a0e.png)

Reply:
**I made your suggestion, the output looks like this.**
![image](https://user-images.githubusercontent.com/66208963/187010875-dec24c66-5cf8-4cc3-8fed-abd214732b7a.png)